src/johnsession.cpp

```diff
@@ -10,6 +10,22 @@ JohnSession::JohnSession(const QString &sessionName, QSettings* settings)
     m_limitSalts = -1;
     m_forkProcesses = 1;
     m_openMPThreads = 0;
+    m_minPasswordCandidatesLength = -1;
+    m_maxPasswordCandidatesLength = -1;
+    m_minMarkovLevel = -1;
+    m_maxMarkovLevel = -1;
+    m_startIndex = -1;
```

Comment:
It would be good if we start indexing mode specific vars with the mode name. m_markovMinLevel m_markovStartIndex